heartbeat/beater/manager.go

```diff
@@ -72,7 +71,7 @@ type monitorTaskConfig struct {
 }
 
 type jobControl interface {
-	Add(sched scheduler.Schedule, name string, f scheduler.TaskFunc) func() error
+	Add(sched scheduler.Schedule, name string, f scheduler.TaskFunc) (stopFn func() error, err error)
```

Comment:
Small note, no need to address this in this PR.

It's more like cancelFn. Idea was to have some kind of handle which allows you to control the job once it is registered. I regret for having a function return here. We should introduce an interface returning a well defined Handle (e.g. `type Handle interface { Cancel(); ... }` )